extensions/amp-o2-player/0.1/amp-o2-player.js

```diff
@@ -31,28 +31,41 @@ class AmpO2Player extends AMP.BaseElement {
   }
 
   /** @override */
-  layoutCallback() {
+  buildCallback() {
     const width = this.element.getAttribute('width');
     const height = this.element.getAttribute('height');
-    const pid = this.element.getAttribute('data-pid');
-    const bcid = this.element.getAttribute('data-bcid');
+
+    /** @private @const {number} */
+    this.width_ = getLengthNumeral(width);
+
+    /** @private @const {number} */
+    this.height_ = getLengthNumeral(height);
+
+    /** @private @const {string} */
+    this.pid_ = user().assert(
+        this.element.getAttribute('data-pid'),
+        'Data-pid attribute is required for <amp-o2-player> %s',
+        this.element);
+
+    /** @private @const {string} */
+    this.bcid_ = user().assert(
+        this.element.getAttribute('data-bcid'),
+        'Data-bcid attribute is required for <amp-o2-player> %s',
+        this.element);
+
     const bid = this.element.getAttribute('data-bid');
     const vid = this.element.getAttribute('data-vid');
     const macros = this.element.getAttribute('data-macros');
     const env = this.element.getAttribute('data-env');
-    user().assert(
-        (pid && bcid) || vid,
-        'Either data-pid and data-bcid or data-vid attribute is required ' +
-        'for <amp-o2-player> %s',
-        this.element);
+
     /** @private {string} */
     this.domain_ = 'https://delivery.' +
-      (env != 'stage' ? '' : 'dev.') + 'vidible.tv';
+        (env != 'stage' ? '' : 'dev.') + 'vidible.tv';
     let src = `${this.domain_}/htmlembed/`;
     const queryParams = [];
-    if (pid && bcid) {
-      src += 'pid=' + encodeURIComponent(pid) + '/'
-        + encodeURIComponent(bcid) + '.html';
+    if (this.pid_ && this.bcid_) {
```

Comment:
This conditional is always true now, since we assert that we have `pid_` and `bcid_` above.